### DISCUSSION TITLE
Does one have to use field.id from useFieldArray as key?

### DISCUSSION THREAD

**Discussion by wondering639** [react-hook-form/react-hook-form#5184] [Q&A] [answered]
When using `useFieldArray` and rendering its contents, do I have to use the id generated by `react-hook-form` (`field.id`) or do I just have to use some unique id? I am especially confused about this from the [docs](https://react-hook-form.com/api/usefieldarray):

> Important: **Because each input can be uncontrolled**, id is required with mapped components to help React to identify which items have changed, are added, or are removed.

How is the React key or field.id related to uncontrolled inputs? Why are uncontrolled components mentioned here at all? Is there more behind `field.id` than being unique and preventing unnnecessary re-renders? Can I provide my own id (something like `product_id`) as key or do I really have to use `field.id`?

**Comment by bluebill1049** [comment, marked as answer]
you can provide your own unique id.

**Reply by wondering639** [reply]
Thanks, glad to hear that. But what about `keyName` of `useFieldArray`. Shall it point to the name of my own id property? What if I just use `key={fields.myKeyProperty}` without setting `keyName: "myKeyProperty"` ?
